CMakeLists.txt

```diff
@@ -2,9 +2,22 @@ cmake_minimum_required(VERSION 2.8.3)
 project(eigen_stl_containers)
 
 find_package(catkin REQUIRED)
+
+# we don't build anything here, but in order to export the right
+# build flags in catkin_package(), we still have to find Eigen3 here
+find_package(Eigen3 REQUIRED)
+
+# eigen 3.2 (willy) only provdies EIGEN3_INCLUDE_DIR, not EIGEN3_INCLUDE_DIRS
+if(NOT EIGEN3_INCLUDE_DIRS)
+  set(EIGEN3_INCLUDE_DIRS ${EIGEN3_INCLUDE_DIR})
+endif()
+if(NOT EIGEN3_LIBRARIES)
+  set(EIGEN3_LIBRARIES)
+endif()
```

Comment:
You don't have to define the libraries variable. `catkin` only complains if *both* variables are undefined - if one is defined it won't print a warning (see https://github.com/ros/catkin/blob/54b7beb709d46c2f5cfb747901907901e954d51e/cmake/catkin_package.cmake#L165-L167).